### PR TITLE
CLN: collect fastpath in Series.__init__

### PR DESCRIPTION
**Pull request by topper-123**
Collect fastpath functionality in one place simplifies implementation

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -378,103 +377,99 @@ def __init__(
...
             if copy:
                 data = data.copy()
-            if index is None:
-                index = data.index
-
-        else:
+            # skips validation of the name
+            object.__setattr__(self, "_name", name)
+            NDFrame.__init__(self, data)
+            return
```
Comment (topper-123) [inline pandas/core/series.py:383]: fastpath now ends here and everything below can simply be moved one level to the left.

**Comment by topper-123** [comment]
The failed check seems unrelated.

File: pandas/core/series.py [hunk 2]
```diff
@@ -378,103 +377,99 @@ def __init__(
                     data = SingleArrayManager.from_array(data, index)
             if copy:
                 data = data.copy()
-            if index is None:
```
Comment (mroeschke) [inline pandas/core/series.py:381]: Why is this `if index is None:` check no longer needed in the `fastpath`?

Comment (topper-123) [reply pandas/core/series.py:381]: If you follow the code all the way down to the end of the method you can see that this actually is a noop; It's superflous code and can be removed. It was just difficult to see in the old code.

Comment (mroeschke) [reply pandas/core/series.py:381]: Ah okay I see it now. Thanks

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @topper-123

**Pull request merged**